msg-socket/src/rep/driver.rs

```diff
@@ -28,6 +32,7 @@ pub(crate) struct PeerState<T: AsyncRead + AsyncWrite> {
     egress_queue: VecDeque<reqrep::Message>,
     state: Arc<SocketState>,
     should_flush: bool,
+    compressor: Option<Arc<dyn Compressor>>,
```

Comment:
Having the compressor be cloned for every new PeerState is not good.

But the PeerState Stream is the only place where incoming messages from a peer
can be compressed, right?